docs/api/security/lifecycle/psa_lifecycle.md

```diff
@@ -0,0 +1,28 @@
+## PSA lifecycle
+
+
+### Description
+
+PSA Lifecycle allows fine grained control over target RoT without compromising on developer experience.
```

Comment:
What exactly do you mean by "without compromising on the developer experience"?

Reply:
we have a typical dilemma:
![dilema](http://static1.squarespace.com/static/531e0c31e4b0f741e5df2ce2/t/53ff3b4ce4b085ab0d3b2a04/1409235789720/21ceb9f.jpg?format=750w).
We need to allow certain functionality (e.g. deleting PSA root of trust, which contains crypto keys and other secrets stored in PSA internal storage), but it weakens solution security. Without this functionality the system is not really usable - e.g. we cannot connect it to CI as we cannot delete all the secretes created during previous session and start from clean state.
This is where lifecycle comes very handy, as it lowers the security bar in favour of usability and functionality while in "lower" lifecycle states (e.g. factory and test)
Without this feature developer experience would be awful: it will allow to run test on a device only once. After that you will need to connect external debugger to for reseting device state (cleaning non volatile memory).
Another example:
In case/once secure bootloader will be deployed - developer may be locked out from secure core as debugger connection is not allowed. Very hard to debug the system in such a state. Different lifecycle state have different debugger policies. So developer will not feel any limitations while device is at factory and test state.

Reply:
Thanks for the clarification! :)

Reply:
@alzix, could you please go over the small update that I made to this document based on your explanation?

Reply:
@alzix, do you have a link to the class reference for this API?